### PR TITLE
fix gradle task dependency issues

### PR DESCRIPTION
**Pull request by NyCodeGHG**
## Proposed Changes

This PR fixes some gradle task dependency issues experienced in #1527.
The maestro-{app,server}.apk files now also get correctly rebuilt when the source has been changed.

Without this PR these warnings get printed, which can lead to wrong results of the build:
```
> Task :maestro-client:processResources
Execution optimizations have been disabled for task ':maestro-client:processResources' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/marie/work/maestro/maestro-client/src/main/resources'. Reason: Task ':maestro-client:processResources' uses this output of task ':maestro-android:copyMaestroAndroid' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.6/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :maestro-studio:server:processResources
Execution optimizations have been disabled for task ':maestro-studio:server:processResources' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/marie/work/maestro/maestro-studio/server/src/main/resources'. Reason: Task ':maestro-studio:server:processResources' uses this output of task ':maestro-studio:server:copyWebFiles' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.6/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

## Testing

I ran the android advanced flow sample using the `maestro` script in the project root, as well as running the build multiple times.

## Issues Fixed
None

### CODE REVIEW

File: maestro-client/build.gradle [hunk 1]
```diff
@@ -104,3 +104,8 @@ plugins.withId("com.vanniktech.maven.publish") {
 test {
     useJUnitPlatform()
 }
+
+processResources {
+    def maestroAndroid = project(":maestro-android")
+    dependsOn(maestroAndroid.tasks.copyMaestroAndroid, maestroAndroid.tasks.copyMaestroServer)
+}
```
Comment (testifyqa) [inline maestro-client/build.gradle:111]: nit: EOF

**Review by testifyqa — APPROVED** [review]

File: maestro-client/build.gradle (hunk 1 above)
Comment (NyCodeGHG) [reply maestro-client/build.gradle]: fixed it!

**Comment by igorsmotto** [comment]
Hi @NyCodeGHG, thanks for this contribution!

Could we also add back the verification of the java version? To make sure we're using the correct one (Java 8)

**Comment by NyCodeGHG** [comment]
@igorsmotto We could do that, but I think it would be better to use [Gradle Toolchains](https://docs.gradle.org/current/userguide/toolchains.html) instead, so it doesn't matter which java version you run gradle with. What do you think?

**Comment by igorsmotto** [comment]
That's an interesting suggestion @NyCodeGHG.

I don't know much about the Toolchain, but if we can make sure that java, kotlin and android compilation tasks will use the same JDK (8) that will definitely works. 

From their docs, it also seems that Toolchain has the option to download the java version if its not autodetected in your local machine, but I believe this can be skipped for now if you want to.

**Comment by NyCodeGHG** [comment]
I changed the project to use toolchains now, maybe it would also be a good idea in the future to reduce code duplication in all of the build.gradle{,.kts} files

**Comment by axelniklasson** [comment]
@NyCodeGHG I'll go ahead and close this PR out as we won't be able to prioritize resources to testing and ensuring that this doesn't break any internal build pipelines that rely on this. Thanks for taking the time to raise this PR!

**Pull request closed without merging**